x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/AsyncSearchUser.java

```diff
@@ -33,6 +33,11 @@
 
     private AsyncSearchUser() {
         super(NAME, ROLE_NAME);
+        // the following traits, and especially the run-as one, go with all the internal users
+        // TODO abstract in a base `InternalUser` class
```

Comment:
I like this. We should do it. Separate PR is ok though.

Reply:
: Thanks. yeah, I had to stop somewhere, this PR spread out quite a bit already.